### DISCUSSION TITLE
How use global store instances to dynamically fetch data in NextJs (getStaticProps / getStaticPaths)?

### DISCUSSION THREAD

**Discussion by alioshr** [vercel/next.js#17501] [Help] [answered]
I am new to Nextjs and I am now wondering what is the best approach in order to go to a dynamic page and getStaticProps from there.

I have these buttons per product snippet on a products page. Once the use clicks on the button it should take him to the products details page ```/pages/products/[id].js```, while passing the products ```id``` of the clicked product into ```redux``` in order to know what to fetch on ```[id].js``` using ```getStaticProps```.

I don't know what to do in order to get this ```id``` passed using ```redux```;

In order to find this nested product in my ```database``` I have two helper instances in my ```productsReducer.js```:

```activeCatalog: "some catalog"```

```activeCategory: "some category related to the active catalog"```

so when I click on this info button in ```Products.js``` I would pass the product ```id``` to ```redux```, then it would be incredibly ease to ```fetch``` it from the ```database```, i.e: ```catalog[store.activeCatalog][store.activeCategory][idFromProduct]```

How can I access the store inside ```getStaticProps``` so I can pass these 3 instances from my ```productsReducer``` into my ```http request```?

Products.js
```
//here is the link to my dynamic page
const IconPanel = ({path}) => (
  <Icons>
    <Link href={`/product/${encodeURIComponent(path)}`}>
    <IconInner>
      <FontAwesomeIcon icon={faPlusCircle} size="lg" />
      <IconSpan>Info</IconSpan>
    </IconInner>
    </Link>
    <FontAwesomeIcon icon={faShoppingBag} size="2x" />
  </Icons>
);
```
This is what I would do using client side http requests, for example:

[id].js - Product Details page

```
import React, {useEffect} from "react";
import {useStore} from 'react-redux'

const ProductDetails = () => {
  const store = useStore(state => state.products)  

  useEffect(() => {
     fetch(`https://meli-345234234.firebaseio.com/catalog/${store.activeCatalog}/${store.activeCatagory}/${store.activeProduct}.json`)
     .then(res => res.json())
     .then(data => console.log(data))
  }, [])

  return <div>Product Detail!</div>;
};
```
How could I get this dynamic URL placed with getStaticProps / getStaticPaths

@ [id].js

```
//ProductDetails function...

export async function getStaticProps() {
    //I wanna have the dynamic route placed on the URL string below
    const res = await fetch('https://meli-345234234.firebaseio.com/catalog.json')
    const number = await res.json()
    return {
        props: {
            number: number
        }
    }
  }

  export async function getStaticPaths() {
     //must repeat this here as I am using a dynamic route
    const res = await fetch('https://meli-345234234.firebaseio.com/catalog.json')
    const data = await res.json()
    return {
        paths: [
            { params: { pid: '1' } },
            { params: { pid: '2' } }
          ],
      fallback: true 
    };
  }
```
I am thinking it out in a wrong way, regarding the use of redux and ```getStaticProps```? Should my case be handled on the client instead? I guess that there will be cases where getStaticProps should just load a shell of the page, for SEO purposes, leaving the other http requests to be done on the client side, as on this example of mine? 

Thanks to all, and sorry for my dummy question!

**Comment by alioshr** [comment, marked as answer]
Now I understand that the only way is by parsing url queries, as REDUX is limited to the client. Please close this discussion.

I am now refining my needs through #17529

**Comment by stegano** [comment]
@alioshr I have written an RFC to solve a problem similar to yours. I hope this helps.
- https://github.com/vercel/next.js/discussions/17657
